packages/react/src/components/Table/TableHeader.tsx

```diff
@@ -1,21 +1,64 @@
-import React, { ThHTMLAttributes } from 'react';
+import React, { useRef, useEffect, ThHTMLAttributes } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import Icon from '../Icon';
+
+// these match aria-sort's values
+type SortDirection = 'ascending' | 'descending' | 'none';
 
 interface TableHeaderProps extends ThHTMLAttributes<HTMLTableCellElement> {
   children: React.ReactNode;
+  sortDirection?: SortDirection;
+  onSort?: () => void;
   className?: string;
 }
 
-const TableHeader = ({ children, className, ...other }: TableHeaderProps) => (
-  <th className={classNames('TableHeader', className)} {...other}>
-    {children}
-  </th>
-);
+const TableHeader = ({
+  children,
+  sortDirection,
+  onSort,
+  className,
+  ...other
+}: TableHeaderProps) => {
+  const sortButtonRef = useRef<HTMLButtonElement>(null);
+  useEffect(() => {
+    if (!sortDirection || sortDirection === 'none') {
+      return;
+    }
+    sortButtonRef.current?.focus();
+  }, [sortDirection]);
+  return (
+    <th
+      aria-sort={sortDirection}
+      className={classNames('TableHeader', className, {
+        'TableHeader--sorting': sortDirection && sortDirection !== 'none'
```

Comment:
I don't think we would want different styles, but might be nice to have the full state here:

* ascending: `TableHeader--sort-ascending`
* descending: `TableHeader--sort-descending`